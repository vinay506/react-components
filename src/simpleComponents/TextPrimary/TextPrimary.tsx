import React from 'react';
import styled from 'styled-components';
function TextPrimary(props: any) {
    /**
     * Text: The text which should be displayed on web page 
     * color: Defaults to black, To override the existing color we can pass Color code or color name as props.
     */

    return (
        <React.Fragment>
            <Label id="textPrimary" style={props.styles}>
                {props.text}
            </Label>
        </React.Fragment >
    )
}

const Label = styled.div`
    color: black;
    font-size: 21px;
    font-family: 'AmazonEmber-Bold';
     letter-spacing: 0;
  line-height: 28px;
    `;
export default TextPrimary;
