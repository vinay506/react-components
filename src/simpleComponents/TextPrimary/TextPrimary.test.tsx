import React from 'react';
import TextPrimary from './TextPrimary';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import 'jest-styled-components'

describe('Text Primary', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultStyles: any
    beforeEach(() => {
        propInfo = {
            text: 'Sample Text Primary',
            styles: {
                color: 'red'
            }
        };
        defaultStyles = {
            color: 'black',
            fontSize: '24px',
            fontFamily: 'AmazonEmber-Bold',
            letterSpacing: '0',
            lineHeight: '28px'
        }
        wrapper = shallow(<TextPrimary {...propInfo} />);
    });
    test('Primary Text Snapshot', () => {
        const instance = renderer.create(<TextPrimary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Primary Text Element check', () => {
        const elementLength = wrapper.find('#textPrimary').length;
        expect(elementLength).toEqual(1);
    })
    test('Primary Text Validation', () => {
        const primaryText = wrapper.find('#textPrimary').text();
        expect(primaryText).toEqual(propInfo.text);
    })
    test('Primary Text Color Validation', () => {
        const primaryTextStyles = wrapper.find('#textPrimary');
        expect(primaryTextStyles.props().style.color).toEqual(propInfo.styles.color);
    })

    // Default Styles check
    test('Text Primary Default Color Validation', () => {
        const instance = renderer.create(<TextPrimary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.color);
    })

    test('Text Primary Default font-size Validation', () => {
        const instance = renderer.create(<TextPrimary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontSize);
    })

    test('Text Primary Default font-family Validation', () => {
        const instance = renderer.create(<TextPrimary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontFamily);
    })

    test('Text Content letterSpacing Validation', () => {
        const instance = renderer.create(<TextPrimary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.letterSpacing);
    })

    test('Text Content lineHeight Validation', () => {
        const instance = renderer.create(<TextPrimary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.lineHeight);
    })
})

