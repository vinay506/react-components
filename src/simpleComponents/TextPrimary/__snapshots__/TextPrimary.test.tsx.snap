// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Primary Primary Text Snapshot 1`] = `
.c0 {
  color: black;
  font-size: 21px;
  font-family: 'AmazonEmber-Bold';
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 28px;
}

<div
  className="c0"
  id="textPrimary"
  style={
    Object {
      "color": "red",
    }
  }
>
  Sample Text Primary
</div>
`;
